pkg/api/source/grpc_adapter.go

```diff
@@ -7,14 +7,13 @@ import (
 
 	"github.com/hashicorp/go-plugin"
 	"google.golang.org/grpc"
-	"google.golang.org/protobuf/types/known/emptypb"
 
 	"github.com/kubeshop/botkube/pkg/api"
 )
 
 // Source defines the Botkube source plugin functionality.
 type Source interface {
-	Stream(ctx context.Context) (StreamOutput, error)
+	Stream(ctx context.Context, configs [][]byte) (StreamOutput, error)
```

Comment:
as the `[][]byte` is used if a few places, I was thinking whether to introduce a dedicated type like:
- `type Configuration []byte` so in func signature: `Stream(ctx context.Context, configs []plugin.Configuration)`
- or - `type Configurations [][]byte` so in func signature: `Stream(ctx context.Context, configs plugin.Configurations)`

WDYT? I would require to stop using the `Request` `Response` on Executor plugin. Is it worth it? On the other hand, it will unify the approach and make the API more self-descriptive.

Reply:
Would sending `repeated Config` over network instead of `repeated bytes` also make sense?

Reply:
Not sure if I understand that correctly. In my comment I only mentioned the Go implementation where we can do simply type aliasing, I don't think that is also supported by protobufs.

Reply:
After a f2f discussion:
```
message Config {
   RawYAML bytes = 1
}

repeated Config
```

so in Go: `config.RawYAML`.